### ISSUE TITLE
ofelia on Debian/buster

### ISSUE THREAD

**Issue by umlaeute** [cuinjune/ofxOfelia#11]
can we expect a binary release for Debian/buster?

the problem being, that the ofelia releases link to some outdated libraries (libglew2.0, libboost-filesystem1.65.1, libboost-system1.65.1), which are no longer available in Debian/buster.

afaict, this also affects users of Ubuntu/disco (19.04) and onwards.

(my filesystem is full of devtools, but i'm somewhat reluctant to install the full OF sdk for playing around)

**Comment by cuinjune** [comment]
@umlaeute I'm not sure at the moment but I can try building ofelia on Debian/buster after its stable release. Please leave a comment here again after this happens.

**Comment by umlaeute** [comment]
buster has been released last saturday...

**Comment by salkin-mada** [comment]
running debian buster - swaywm - rt kernel
wanted to give ofelia a go today.
But I am not able to run any ofelia examples. Pd not so happy.
```
 ldd ofelia.pd_linux | grep not        
        libglfw.so.3 => not found
        libGLEW.so.2.0 => not found
        libboost_filesystem.so.1.65.1 => not found
        libboost_system.so.1.65.1 => not found
        liburiparser.so.1 => not found
```
`sudo ./Pd/externals/ofelia/scripts/debian/install_dependencies.sh`
is not solving my problem, i guess this is directly related to OP @umlaeute 
debian buster is no longer using these outdated libraries.

an update in the making or did i miss something?

**Comment by cuinjune** [comment]
I think you need to build the external yourself.

https://forum.openframeworks.cc/t/rpi-4-raspbian-buster-openframeworks/32856/59

Install openFrameworks, download [ofxOfelia_Linux](https://github.com/cuinjune/Ofelia/releases/download/v3.1.0/ofxOfelia_Linux.zip)

Unzip,  rename(to ofxOfelia) and copy the folder to "OF/addons" folder.
cd to "ofxOfelia/LinuxExternal" directory and type make to build the external.
You will find the generated binary "ofelia.l_arm" inside "ofxOfelia/LinuxExternal/bin" folder.
Copy and replace it with the existing one.

**Comment by cuinjune** [comment]
This is how I could build ofelia on Rapbian Buster.

Install openFrameworks 0.11.0 following the link in the above comment.

In `ofxOfelia/libs/ofxLua/src/bindings/glmbindings.cpp`,
comment out the following line:
`#include <glm/gtc/functions.hpp>`

Find and open `config.make` in both `LinuxExample` and `LinuxExternal` and then find and change
`-Os` to `-O2`

Now compile using `make`

**Comment by salkin-mada** [comment]
Perfecto, thanks @cuinjune. Will try asap. And good to hear an update is in the making.

**Comment by umlaeute** [comment]
thanks for the detailed instructions on how to build ofelia yourself.

but to re-ask my original question:
> can we expect a binary release for Debian/buster?

:wink:

**Comment by ingox--** [comment]
I could make Ofelia run on debian buster by changing in scripts/debian/install_dependencies.sh on line 40: libgles1-mesa-dev to libglvnd-dev

and then doing:
sudo ln -s libGLEW.so.2.1.0 libGLEW.so.2.0
sudo ln -s libboost_filesystem.so.1.67.0 libboost_filesystem.so.1.65.1
sudo ln -s libboost_system.so.1.67.0 libboost_system.so.1.65.1

After that it seems to work fine. :)

Thank you, @cuinjune!

**Comment by umlaeute** [comment]
this way lies madness.

seriously, while your symlinks might make things work, they can **severely** break other applications.
after all, there's a very good reason why those dynamic libraries have their names changed.

(it might work for you alright; but the internet is full of awful advise that people blindly follow without having a clue what they are actually doing - rendering their systems unusable, and then claiming that "linux is just broken")

**Comment by ingox--** [comment]
@umlaeute True. I reported this so the installation files could possibly be adapted to use the updated libraries. Then Ofelia could simply installed via Deken and would work on Buster.
